src/frontend/src/components/deviceListItem.ts

```diff
@@ -1,17 +1,32 @@
 import { TemplateResult, html } from "lit-html";
 import { Setting, settingName } from "../flows/manage/deviceSettings";
-import { warningIcon, dropdownIcon, lockIcon } from "./icons";
+import {
+  warningIcon,
+  errorIcon,
+  successIcon,
+  dropdownIcon,
+  lockIcon,
+} from "./icons";
 
 // A simple representation of "device"s used on the manage page.
 export type Device = {
   // All the settings allowed for a particular device
-  settings: Setting[];
+  settings?: Setting[];
   // The displayed name of a device (not exactly the "alias") because
   // recovery devices handle aliases differently.
-  label: string;
-  isRecovery: boolean;
-  isProtected: boolean;
-  warn?: TemplateResult;
+  label: string | TemplateResult;
+  recovery?: "phrase" | "device";
+  isProtected?: boolean;
+  status?: {
```

Comment:
instead of `status?` wouldn't be safer to have always a `status` but with an initial `statusType: "init"`?

Reply:
these are exceptions, we use this to show some kind of notification or error on each device or recovery method.